feature-libs/asm/customer-360/components/asm-customer-promotion-listing/asm-customer-promotion-listing.component.html

```diff
@@ -3,6 +3,61 @@ <h4 class="cx-asm-customer-promotion-listing-heading-text">
     {{ headerText }}
   </h4>
 </div>
+<ng-container *ngIf="isCustomerCoupon">
+  <div class="cx-asm-customer-promotion-listing-tabs">
+    <button
+      class="cx-tab-header"
+      [class.active]="activeTab === 0"
+      [textContent]="'customer360.customerCoupons.availableTab' | cxTranslate"
+      (click)="
+        refreshCustomerCouponList.emit({
+          assignable: true,
+          searchQuery: undefined
+        });
+        activeTab = 0
+      "
+    ></button>
+    <button
+      class="cx-tab-header"
+      [class.active]="activeTab === 1"
+      [textContent]="'customer360.customerCoupons.sentTab' | cxTranslate"
+      (click)="
+        refreshCustomerCouponList.emit({
+          assignable: false,
+          searchQuery: undefined
+        });
+        activeTab = 1
+      "
+    ></button>
+  </div>
+  <hr class="cx-asm-customer-promotion-listing-separator" aria-hidden="true" />
+  <div class="cx-asm-customer-promotion-listing-search">
+    <input
+      #searchBox
+      class="cx-asm-customer-promotion-listing-search-input"
+      placeholder="{{ 'customer360.customerCoupons.searchBox' | cxTranslate }}"
+      (keydown.enter)="
```

Comment:
if value changed do we need tigger search ? if yes maybe need consider debounce , if not ignore it.

Reply:
only click enter or the search button will trigger search.